generator/src/main/java/com/linkedin/pegasus/generator/DataSchemaParser.java

```diff
@@ -47,31 +50,46 @@
 {
   private final String _resolverPath;
   private final Map<String, FileFormatDataSchemaParser> _parserByFileExtension;
-  private final MultiFormatDataSchemaResolver _resolver;
+  private final AbstractMultiFormatDataSchemaResolver _resolver;
 
   /**
    * @param resolverPath provides the search paths separated by the system file separator, or null for no search paths.
    */
-  public DataSchemaParser(String resolverPath) {
-    this(resolverPath, MultiFormatDataSchemaResolver.BUILTIN_FORMAT_PARSER_FACTORIES);
+  public DataSchemaParser(String resolverPath)
+  {
+    this(resolverPath, AbstractMultiFormatDataSchemaResolver.BUILTIN_FORMAT_PARSER_FACTORIES);
   }
 
   public DataSchemaParser(
       String resolverPath,
       List<DataSchemaParserFactory> parserFactoriesForFromats)
   {
+    _parserByFileExtension = new HashMap<>();
     _resolverPath = resolverPath;
-
-    MultiFormatDataSchemaResolver resolver =
+    AbstractMultiFormatDataSchemaResolver resolver =
```

Comment:
Shall we use MultiFormatDataSchemaResolver type here

Reply:
yeah